server/src/main/java/org/opensearch/snapshots/SnapshotUtils.java

```diff
@@ -69,6 +71,14 @@ public static List<String> filterIndices(List<String> availableIndices, String[]
         if (IndexNameExpressionResolver.isAllIndices(Arrays.asList(selectedIndices))) {
             return availableIndices;
         }
+
+        // Move the exclusions to end of list to ensure they are processed
+        // after explicitly selected indices are chosen.
+        final List<String> excludesAtEndSelectedIndices = Stream.concat(
```

Comment:
You need to replace every usage of `selectedIndices` below with `excludesAtEndSelectedIndices`

Reply:
Ooops--clearly I had vacation brain hah.